packages/web-frontend/src/sagas.js

```diff
@@ -908,47 +907,66 @@ function* fetchMeasureInfo() {
       yield put(fetchMeasureInfoError(error));
     }
   }
-  yield put(fetchAllMeasureInfoSuccess());
+  if (displayedMapOverlays) {
+    yield put(setDisplayedMapOverlays(displayedMapOverlays));
+  }
+  yield put(fetchAllMeasureInfoSuccess(mapOverlayCodes));
 }
 
 function* watchSetMapOverlayChange() {
-  yield takeLatest(SET_MAP_OVERLAYS, fetchMeasureInfo);
+  yield takeLatest(SET_MAP_OVERLAYS, function* _() {
+    const state = yield select();
+    const mapOverlayCodes = selectCurrentMapOverlayCodes(state);
+    const { displayedMapOverlays, measureInfo } = state.map;
+
+    const previousDisplayedOverlays = displayedMapOverlays.filter(code =>
+      mapOverlayCodes.includes(code),
+    );
+    const newSelectedMapOverlays = mapOverlayCodes.filter(code => !measureInfo[code]);
+
+    yield fetchMeasureInfo(mapOverlayCodes, [
+      ...previousDisplayedOverlays,
+      ...newSelectedMapOverlays,
+    ]);
```

Comment:
The filtering logic above seems legit, but would like to have a quick chat with you so that I can fully understand it